### PR TITLE
test: add kvstore basic test app

### PR DESCRIPTION
**Pull request by JakeChampion**
This adds an integration test to show that the happy-paths for kvstore open, insert, and lookup methods work as expected

### CODE REVIEW

**Review by dgryski — APPROVED** [review]
LGTM

**Comment by JakeChampion** [comment]
@dgryski I don't have write access, could you please merge this when you have a moment?

**Pull request merged**
